src/views/GuestBook.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Box,
+  Grid,
+  Typography,
+  Button,
+  Container,
+  CssBaseline,
+  TextField,
+  Divider,
+  IconButton,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Avatar,
+} from '@mui/material';
+import { IoIosHome } from 'react-icons/io';
+import { useUser } from '../context/UserContext';
+import { Link, useHistory } from 'react-router-dom';
+import { createEntry, getEntries } from '../services/entries';
+import { signOutUser } from '../services/auth';
+import LogoutIcon from '@mui/icons-material/Logout';
+import Delete from '../components/Delete';
+
+function GuestBook() {
+  const { user, setUser } = useUser();
+  const [signature, setSignature] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [entries, setEntries] = useState([]);
+  const [error, setError] = useState('');
+  const [updateEntries, setUpdateEntries] = useState({});
+  const history = useHistory();
+  useEffect(() => {
+    getEntries()
+      .then(setEntries)
+      .catch(setError)
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    setLoading(true);
+    getEntries()
+      .then(setEntries)
+      .catch(setError)
+      .finally(() => setLoading(false));
+  }, [updateEntries]);
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const newEntry = await createEntry({ userId: user.id, signature });
+    setUpdateEntries(newEntry);
+    setSignature('');
+  };
+  const handleSignOut = () => {
+    signOutUser().then(setUser({})).finally(history.push('/'));
+  };
+  return (
+    <Container component="main">
+      <Container component="header">
+        <CssBaseline />
+        <Grid container direction="row" justifyContent="space-between">
+          <Grid item>
+            <Link to="/">
+              <IconButton>
+                <IoIosHome />
+              </IconButton>
+            </Link>
+          </Grid>
+          <Grid item>
+            <Delete update={setUpdateEntries} />
+          </Grid>
+          <Grid item>
+            <Button
+              onClick={handleSignOut}
+              variant="outlined"
+              endIcon={<LogoutIcon />}
+            >
+              Logout
+            </Button>
+          </Grid>
+        </Grid>
+        <Typography component="h1" variant="h5">
+          Please sign the guest book!
```

Comment:
You can use the context to render the user's email here to personalize the app to this. You have access with `user.email`